neuraxle/base.py

```diff
@@ -1248,7 +1870,28 @@ def _step_index_to_name(self, step_index):
         name, _ = self.steps_as_tuple[step_index]
         return name
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: Union[slice, int, str]):
+        """
+        Truncate self with a slice, an index or a step name.
+
+        Example :
+        ```
+        p = Pipeline([
+            ('1', SomeStep()),
+            ('2', SomeStep()),
+            ('3', SomeStep())
+        ])
+        p[0] # returns the first SomeStep()
+        p[0:2] # returns a TruncableSteps containing the first, and second SomeStep()
+        p['2'] # returns the second SomeStep()
+        ```
+
+        :param key: slice, index, or step name
+        :type key: Union[slice, int, str]
+
+        :return: truncated self
+        :rtype: Union[TruncableSteps, BaseStep]
```

Comment:
I would add a link to this stack overflow answer and/or include the example I gave on stack overflow: 

.. See also:  https://stackoverflow.com/questions/28822756/getting-model-attributes-from-scikit-learn-pipeline/58359509#58359509